pkg/alertmanager/alertmanager.go

```diff
@@ -47,18 +48,13 @@ func New(cfg pkgconfig.Config) Service {
 	return service
 }
 
-func ExtractAlertmanagerConfig(ctx context.Context, secret *v1.Secret) ([]byte, error) {
+// ExtractAlertmanagerConfig extracts the raw config bytes.
+func ExtractAlertmanagerConfig(secret *v1.Secret) ([]byte, error) {
 	// Check that the secret contains an Alertmanager configuration file.
 	alertmanagerConfig, found := secret.Data[AlertmanagerConfigKey]
 	if !found {
 		return nil, fmt.Errorf("missing %s in alertmanager secret", AlertmanagerConfigKey)
 	}
-	// Validate Alertmanager configuration
-	// The returned config is not used, as transforming it via String() would produce an invalid configuration with all secrets replaced with <redacted>.
```

Comment:
Please keep this comment as its important that we never send the returned config to alertmanager.